app/helpers/prefill_fields_helper.rb

```diff
@@ -76,13 +126,88 @@ def ats_fields_cache_key(ats_fields_param)
 
   def cache_result(cache_key, value, ttl)
     Rails.cache.write(cache_key, value, expires_in: ttl)
-  rescue StandardError => e
-    Rails.logger.warn "Cache write failed for ATS fields: #{e.message}"
+  rescue StandardError
     # Continue execution even if caching fails
   end
 
   def cache_and_return_empty(cache_key)
     cache_result(cache_key, [], 5.minutes)
     []
   end
+
+  # Builds an optimized lookup cache for field UUID resolution
+  #
+  # Creates a hash mapping ATS field names to template field UUIDs for O(1) lookup
+  # performance instead of O(n) linear search. Results are cached to improve
+  # performance across multiple requests.
+  #
+  # @param template_fields [Array<Hash>, nil] Template field definitions
+  # @return [Hash] Mapping of ATS field names to field UUIDs
+  #
+  # @example
+  #   template_fields = [
+  #     { 'uuid' => 'field-1', 'prefill' => 'employee_first_name' },
+  #     { 'uuid' => 'field-2', 'prefill' => 'employee_last_name' }
+  #   ]
+  #   build_field_lookup_cache(template_fields)
+  #   # => { 'employee_first_name' => 'field-1', 'employee_last_name' => 'field-2' }
+  def build_field_lookup_cache(template_fields)
```

Comment:
Non-blocking: Can we break up this method into a few smaller methods? We have 2 begin rescue blocks and an each_with_object loop.